File: uva_11246.py
num=int(input())
for _ in range(num):
    ans=0
    n, k=input().split()
    n, k=int(n), int(k)
    sign=True
    while n:
        ans+=n if sign else -n
        n=int(n/k)
        sign=False if sign else True
    print(ans)

File: uva_11246.cpp
# include <iostream>
using namespace std;

int main(void){
    int num;
    cin >> num;
    while(num--){
        int n, k, ans=0;
        bool sign=true;
        cin >> n >> k;
        while(n){
            ans+=sign?n:-n;
            n/=k;
            sign=sign?false:true;
        }
        cout << ans << endl;
    }
    return 0;
}